File: framework/languages/java/samples/kotlin/Bingo/Server/Play/src/main/kotlin/systems/zlink/samples/kotlin/bingo/server/play/actors/PlayerActorFactory.kt
package systems.zlink.samples.kotlin.bingo.server.play.actors

import systems.zlink.framework.actors.ZLinkActor
import systems.zlink.framework.actors.ZLinkActorContext
import systems.zlink.framework.kotlin.ZLinkCoroutineActorFactory
import systems.zlink.framework.kotlin.ZLinkCoroutineRuntime

class PlayerActorFactory(
    coroutines: ZLinkCoroutineRuntime,
) : ZLinkCoroutineActorFactory(coroutines) {
    override suspend fun create(actorId: String, context: ZLinkActorContext): ZLinkActor =
        PlayerActor(actorId, context)
}

File: framework/languages/node/samples/ZoneWorld/Server/ZoneNode/Infrastructure/ZLink/Actors/player-actor-factory.ts
import { PlayerActor } from './player-actor';
import type { ZLinkActorContext, ZLinkActorFactory } from '@zlink-systems/framework';

class PlayerActorFactory implements ZLinkActorFactory {
  async create(actorId: string, context: ZLinkActorContext): Promise<PlayerActor> {
    const actor = new PlayerActor(actorId);
    Object.defineProperty(actor, 'context', {
      configurable: true,
      enumerable: false,
      value: context
    });
    return actor;
  }
}

export { PlayerActorFactory };
